app/app.go

```diff
@@ -399,6 +414,13 @@ func NewGaiaApp(
 		),
 	)
 	app.SetEndBlocker(app.EndBlocker)
+	app.UpgradeKeeper.SetUpgradeHandler("Gravity-DEX",
```

Comment:
Do we need a command line flag to allow gaia with new chain and fresh genesis or as an upgrade to comsoshub-4?

Reply:
No flag required, if the chain stops in the upgrade-height specified in the software-upgrade proposal, we can upgrade by only restarting to the new version of Gaia with the liquidity module

Here is a test script for this scenario

```bash
gaiad version
# > v4.2.1

gaiad init testing --chain-id testing

# fix voting period, bond_denom
cd ~/.gaia/config
sed -i '' 's/172800s/10s/' genesis.json
sed -i '' 's/stake/uatom/' genesis.json

gaiad keys add validator --keyring-backend test
gaiad keys add user1 --keyring-backend test
gaiad add-genesis-account $(gaiad keys show validator --keyring-backend test -a) 10000000000stake,10000000000uatom,500000000000uusd
gaiad add-genesis-account $(gaiad keys show user1 --keyring-backend test -a) 10000000000stake,10000000000uatom,500000000000uusd
gaiad gentx validator 1000000000uatom --chain-id testing --keyring-backend test

gaiad collect-gentxs

gaiad start

gaiad tx gov submit-proposal software-upgrade Gravity-DEX -y \
  --deposit 10000000uatom \
  --title Gravity-DEX \
  --upgrade-height 10 \
  --upgrade-info v4.3.0-testhash \
  --from validator \
  --chain-id testing \
  --description testdescription \
  --keyring-backend test

gaiad tx gov vote 1 yes -y --chain-id testing --from validator  --keyring-backend test

# checking the proposal is passed after 10s
gaiad query gov proposal 1


# panic on upgrade-height
# 7:46PM ERR UPGRADE "v4.3.0" NEEDED at height: 10: testhash
# 7:46PM ERR CONSENSUS FAILURE!!! err="UPGRADE \"v4.3.0\" NEEDED at height: 10: testhash"

# try to restart current version of gaia
# 4:26PM ERR UPGRADE "Gravity-DEX" NEEDED at height: 10: v4.3.0-testhash
# panic: UPGRADE "Gravity-DEX" NEEDED at height: 10: v4.3.0-testhash


# build new gaia using [this-branch](https://github.com/b-harvest/gravity-dex/tree/add-liquidity-module-to-gaia)
make build 
cd /build

./gaiad version
# > add-liquidity-module-to-gaia-xxxx

./gaiad start --home ~/.gaia

./gaiad query liquidity params

./gaiad tx liquidity create-pool 1 1000000000uatom,50000000000uusd --from user1 --keyring-backend test --chain-id testing -y
./gaiad tx liquidity swap 1 1 50000000uusd uatom 0.019 0.003 --from validator --chain-id testing --keyring-backend test -y
```